src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -842,50 +842,58 @@ private void EmbedSourceLink(Stream stream)
         /// Capture the set of compilation options to allow a compilation 
         /// to be reconstructed from the pdb
         /// </summary>
-        private void EmbedCompilationOptions(CommonPEModuleBuilder module)
+        private void EmbedCompilationOptions(BlobReader? pdbCompilationOptionsReader, CommonPEModuleBuilder module)
         {
             var builder = new BlobBuilder();
 
-            var compilerVersion = typeof(Compilation).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
-            WriteValue(CompilationOptionNames.CompilationOptionsVersion, CompilationOptionsSchemaVersion.ToString());
-            WriteValue(CompilationOptionNames.CompilerVersion, compilerVersion);
-
-            WriteValue(CompilationOptionNames.Language, module.CommonCompilation.Options.Language);
-            WriteValue(CompilationOptionNames.SourceFileCount, module.CommonCompilation.SyntaxTrees.Count().ToString());
-
-            if (module.EmitOptions.FallbackSourceFileEncoding != null)
+            if (pdbCompilationOptionsReader is { } reader)
             {
-                WriteValue(CompilationOptionNames.FallbackEncoding, module.EmitOptions.FallbackSourceFileEncoding.WebName);
+                // TODO: nervous about this. how can we compare the "real" compilation options when things fail in spite of this.
```

Comment:
That doesn't make me nervous, it's the responsibility of the host. The host already has to read the options blob in order to setup the compilation, check the environment, etc ... That is where any diffing of the options or diagnostics like 

> Using compiler 3.8.2 instead of 3.8.1 

Can live

Reply:
Removed the TODO